src/test/java/io/strimzi/kafka/bridge/http/services/ConsumerService.java

```diff
@@ -0,0 +1,193 @@
+ */
+package io.strimzi.kafka.bridge.http.services;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.strimzi.kafka.bridge.BridgeContentType;
+import io.strimzi.kafka.bridge.utils.Urls;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.HttpRequest;
+import io.vertx.ext.web.client.HttpResponse;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.codec.BodyCodec;
+import io.vertx.junit5.VertxTestContext;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static com.google.common.net.HttpHeaders.ACCEPT;
+import static com.google.common.net.HttpHeaders.CONTENT_LENGTH;
+import static com.google.common.net.HttpHeaders.CONTENT_TYPE;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ConsumerService extends BaseService {
+
+    public ConsumerService(WebClient webClient) {
+        super(webClient);
+    }
+
+    // Consumer basic requests
+
+    public HttpRequest<JsonObject> createConsumerRequest(String groupId, JsonObject json) {
+        return postRequest(Urls.consumer(groupId))
+                .putHeader(CONTENT_LENGTH, String.valueOf(json.toBuffer().length()))
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON);
+    }
+
+    HttpRequest<JsonObject> deleteConsumerRequest(String url) {
+        return deleteRequest(url)
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON)
+                .as(BodyCodec.jsonObject());
+    }
+
+    public HttpRequest<Buffer> consumeRecordsRequest(String groupId, String name) {
+        return getRequest(Urls.consumerInstancesRecords(groupId, name, 1000, null))
+                .putHeader(ACCEPT, BridgeContentType.KAFKA_JSON_JSON);
+    }
+
+    public HttpRequest<JsonObject> subscribeConsumerRequest(String groupId, String name, JsonObject json) {
+        return postRequest(Urls.consumerInstancesSubscription(groupId, name))
+                .putHeader(CONTENT_LENGTH, String.valueOf(json.toBuffer().length()))
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON);
+    }
+
+    public HttpRequest<JsonObject> offsetsRequest(String groupId, String name, JsonObject json) {
+        return postRequest(Urls.consumerInstancesOffsets(groupId, name))
+                .putHeader(CONTENT_LENGTH, String.valueOf(json.toBuffer().length()))
+                .putHeader(CONTENT_TYPE, BridgeContentType.KAFKA_JSON)
+                .as(BodyCodec.jsonObject());
+    }
+
+    // Consumer actions
+
+    public ConsumerService deleteConsumerSubscription(VertxTestContext context, String groupId, String name, String... topicNames) throws InterruptedException, ExecutionException, TimeoutException {
```

Comment:
technically it's an "unsubscribe" request, I would not call it as a "delete". Anyway, it's going to use the HTTP DELETE method of course.